pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -0,0 +1,504 @@
+    Schema avroSchema = null;
+    for (String input : inputPathDir.split("\\s*,\\s*")) {
+      Path inputPath = new Path(input);
+      for (FileStatus fileStatus : fs.listStatus(inputPath)) {
+        if (fileStatus.isFile() && fileStatus.getPath().getName().endsWith(".avro")) {
+          _logger.info("Extracting schema from " + fileStatus.getPath());
+          avroSchema = extractSchemaFromAvro(fileStatus.getPath());
+          break;
+        }
+      }
+    }
+    return avroSchema;
+  }
+
+  /**
+   * Extracts avro schema from avro file
+   * @param avroFile Input avro file
+   * @return Schema in avro file
+   * @throws IOException exception when accessing to IO
+   */
+  private static Schema extractSchemaFromAvro(Path avroFile) throws IOException {
+    DataFileStream<GenericRecord> dataStreamReader = getAvroReader(avroFile);
+    Schema avroSchema = dataStreamReader.getSchema();
+    dataStreamReader.close();
+    return avroSchema;
+  }
+
+  /**
+   * Helper method that returns avro reader for the given avro file.
+   * If file name ends in 'gz' then returns the GZIP version, otherwise gives the regular reader.
+   *
+   * @param avroFile File to read
+   * @return Avro reader for the file.
+   * @throws IOException exception when accessing to IO
+   */
+  private static DataFileStream<GenericRecord> getAvroReader(Path avroFile) throws IOException {
+    FileSystem fs = FileSystem.get(new Configuration());
+    if (avroFile.getName().endsWith("gz")) {
+      return new DataFileStream<>(new GZIPInputStream(fs.open(avroFile)), new GenericDatumReader<>());
+    } else {
+      return new DataFileStream<>(fs.open(avroFile), new GenericDatumReader<>());
+    }
+  }
+
+  private ArrayList<FileStatus> getDataFilesFromPath(FileSystem fs, Path inBaseDir) throws IOException {
+    ArrayList<FileStatus> dataFileStatusList = new ArrayList<>();
+    FileStatus[] fileStatusArr = fs.listStatus(inBaseDir);
+    for (FileStatus fileStatus : fileStatusArr) {
+      if (fileStatus.isDirectory()) {
+        _logger.info("Trying to add all the data files from directory: {}", fileStatus.getPath());
+        dataFileStatusList.addAll(getDataFilesFromPath(fs, fileStatus.getPath()));
+      } else {
+        String fileName = fileStatus.getPath().getName();
+        if (fileName.endsWith(".avro")) {
+          _logger.info("Adding avro files: {}", fileStatus.getPath());
+          dataFileStatusList.add(fileStatus);
+        }
+        if (fileName.endsWith(".csv")) {
+          _logger.info("Adding csv files: {}", fileStatus.getPath());
+          dataFileStatusList.add(fileStatus);
+        }
+        if (fileName.endsWith(".json")) {
+          _logger.info("Adding json files: {}", fileStatus.getPath());
+          dataFileStatusList.add(fileStatus);
+        }
+      }
+    }
+    return dataFileStatusList;
+  }
+
+  private void addContainerLocalFiles(final Job job, final String cacheFolder, final Configuration conf)
```

Comment:
are we using the same distributed cache code that we use elsewhere in segment creation code? do we need to recopy this if so or is this for something else?